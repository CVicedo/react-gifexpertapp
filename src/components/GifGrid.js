import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs'
import { GifGridItem } from './GifGridItem'
import PropTypes from 'prop-types'



export const GifGrid = ({category}) => {

    const {data: images, loading} = useFetchGifs( category )

    const imagesYes = (images) => {
        return(
            <div className="card-grid">
                    {images.map((img) => {
                        return(
                            <GifGridItem
                                key={img.id}
                                {...img}
                            />
                        )
                    })}
            </div>
        )
    }

    const imagesNo = () => <p>Sorry, no gifs found.</p>;
        
    return (
        <>
            <h3 className="animate__animated animate__fadeIn">{category}</h3>

            {loading && <p className="animate__animated animate__flash">Loading...</p>}
            
            {images.length === 0 ? imagesNo() : imagesYes(images) }
            
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}
