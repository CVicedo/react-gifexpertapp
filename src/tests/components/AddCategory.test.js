import React from 'react'
import { AddCategory } from '../../components/AddCategory'
import { shallow } from 'enzyme'

describe('Pruebas en <AddCategory />', () => {

    const setCategories = jest.fn()
    let wrapper = shallow(<AddCategory setCategories={setCategories} />);

    beforeEach(() => {
        jest.clearAllMocks()
        wrapper = shallow(<AddCategory setCategories={setCategories} />);
    })

    test('debe de mostrarse correctamente', () => {
        expect( wrapper ).toMatchSnapshot()
    })

    test('debe cambiar la caja de texto', () => {
        
        const input = wrapper.find('input');
        const value = 'hola mundo'

        input.simulate('change', { target: { value } });

        expect(wrapper.find('p').text().trim()).toBe(value)
    })

    test('NO debe de postear la informaciÃ³n onSubmit', () => {
        
        wrapper.find('form').simulate('submit', { preventDefault(){} })

        expect( setCategories ).not.toHaveBeenCalled()

    })

    test('debe de llamar setcategories y limpiar la caja de texto', () => {
        const value = 'Hola mundo'

        wrapper.find('input').simulate('change', { target: { value } });

        wrapper.find('form').simulate('submit', {preventDefault(){}});

        expect(setCategories).toHaveBeenCalled();
        expect(setCategories).toHaveBeenCalledWith( expect.any(Function) );
        
        expect(wrapper.find('input').prop('value')).toBe('')

    })
    
    
    

})