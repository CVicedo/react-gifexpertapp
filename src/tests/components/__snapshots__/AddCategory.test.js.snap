// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AddCategory /> debe de mostrarse correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    placeholder="Search..."
    type="text"
    value=""
  />
</form>
`;
